ql/src/java/org/apache/hadoop/hive/ql/parse/MetaDataExportListener.java

```diff
@@ -90,11 +92,9 @@ private void export_meta_data(PreDropTableEvent tableEvent) throws MetaException
       EximUtil.createExportDump(fs, outFile, mTbl, null, null,
           new HiveConf(conf, MetaDataExportListener.class));
       if (moveMetadataToTrash == true) {
-        wh.deleteDir(metaPath, true);
+        wh.deleteDir(metaPath, true, Hive.get().getDatabase(tbl.getDbName()));
```

Comment:
Here we are deleting the metaPath which is the export dir not warehouse location. So, last argument should be false always.